featuretools/entityset/entityset.py

```diff
@@ -526,403 +556,459 @@ def entity_from_dataframe(self,
-                raise ValueError("'make_time_index' must be specified in 'additional_variables' or 'copy_variables'")
-        if index == base_entity.index:
-            raise ValueError("'index' must be different from the index column of the base entity")
+            if make_time_index not in base_dataframe.columns:
+                raise ValueError("'make_time_index' must be a column in the base dataframe")
+            elif make_time_index not in additional_columns + copy_columns:
+                raise ValueError("'make_time_index' must be specified in 'additional_columns' or 'copy_columns'")
+        if index == base_dataframe.ww.index:
+            raise ValueError("'index' must be different from the index column of the base dataframe")
 
         transfer_types = {}
-        transfer_types[index] = type(base_entity[index])
-        for v in additional_variables + copy_variables:
-            if type(base_entity[v]) == vtypes.DatetimeTimeIndex:
-                transfer_types[v] = vtypes.Datetime
-            elif type(base_entity[v]) == vtypes.NumericTimeIndex:
-                transfer_types[v] = vtypes.Numeric
-            else:
-                transfer_types[v] = type(base_entity[v])
-
-        # create and add new entity
-        new_entity_df = self[base_entity_id].df.copy()
-
-        if make_time_index is None and base_entity.time_index is not None:
+        # Types will be a tuple of (logical_type, semantic_tags, column_metadata, column_description)
+        transfer_types[index] = (base_dataframe.ww.logical_types[index],
+                                 base_dataframe.ww.semantic_tags[index],
+                                 base_dataframe.ww.columns[index].metadata,
+                                 base_dataframe.ww.columns[index].description)
+        for col_name in additional_columns + copy_columns:
+            # Remove any existing time index tags
+            transfer_types[col_name] = (base_dataframe.ww.logical_types[col_name],
+                                        (base_dataframe.ww.semantic_tags[col_name] - {'time_index'}),
+                                        base_dataframe.ww.columns[col_name].metadata,
+                                        base_dataframe.ww.columns[col_name].description)
+
+        # create and add new dataframe
+        new_dataframe = self[base_dataframe_name].copy()
+
+        if make_time_index is None and base_dataframe.ww.time_index is not None:
             make_time_index = True
 
         if isinstance(make_time_index, str):
             # Set the new time index to make_time_index.
             base_time_index = make_time_index
-            new_entity_time_index = make_time_index
-            already_sorted = (new_entity_time_index == base_entity.time_index)
+            new_dataframe_time_index = make_time_index
+            already_sorted = (new_dataframe_time_index == base_dataframe.ww.time_index)
```

Comment:
`already_sorted` evaluating to `True` is not tested on main branch, we may want to make an issue to add a test